docs/designs/data_transform.md

```diff
@@ -210,64 +224,103 @@ COLUMN
 
-The [submitter pragram](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/model_zoo.md#submitter-programs) will combine these two and feed the output tensor of transform code into the model definition.  
+#### Derive the parameters of TransformOP from DAG Walkthrough
 
-### Implementation (To Be Improved)
+Some parameters of TransformOP can't be gotten from data analysis. For example, `Embedding` needs the parameter `input_dimension`, it's equals to `num_buckets` attribute of `CONCAT` which is the dependency of `Embedding`.  
+In one sentence, we need walk through the DAG and calculate the parameters of one TransformOP from its dependency.
 
-Data transform contains two stages: analyze and transform. In our design, we will do the analysis using SQL as the first step, and generate the feature column definition as the second step. The feature column contains the transform logic and executes along with the model training process.  
-We choose to convert the transform expression into two steps of the work flow described by [Couler](https://github.com/sql-machine-learning/sqlflow/blob/develop/python/couler/README.md): analyze and feature column generation. Couler is a programming language for describing workflows. Its compiler translates a workflow represented by a Python program into an [Argo](https://argoproj.github.io/) YAML file. The output of feature column generation will be passed to the next model training step.  
-![data_transform_pipeline](../images/data_transform_pipeline.png)
+#### Generate the Transform Python code from the DAG
 
-Let's take STANDARDIZE(age) for example, the following figure describes how the data transform pipeline works in detail.  
+Now we have a complete DAG: a complete Graph describing the dependency of TransformOPs and each TranformOP has all the required parameters.  
+We can make typological sort on the DAG and get an ordered list. Then we generate the python code according to this order. A TransformOP node will generate a line of python code. A line of python code is an api call to [feature column](https://www.tensorflow.org/versions/r2.2/api_docs/python/tf/feature_column) or [keras preprocessing layer](https://github.com/tensorflow/community/pull/188) from TensorFlow native API and our extensions in [elasticdl_preprocessing](https://github.com/sql-machine-learning/elasticdl/tree/develop/elasticdl_preprocessing). Please check the sample code generated from the example SQL: [keras preprocessing layer version](https://github.com/sql-machine-learning/elasticdl/blob/84bf8026565df81521ffdfe55d854428fb1156d4/model_zoo/census_model_sqlflow/wide_and_deep/wide_deep_functional_tensor_interface_keras.py#L129-L228) and [feature column version](https://github.com/sql-machine-learning/elasticdl/blob/84bf8026565df81521ffdfe55d854428fb1156d4/model_zoo/census_model_sqlflow/wide_and_deep/wide_deep_functional_tensor_interface_fc.py#L130-L243).
```

Comment:
We assume to use the order from `SELECT ...` statement now since it's more readable and simple.

Reply:
I'm afraid not.
The order of the python code is the same with the order of the TransformOP execution.
One TransformOP (not an item from the source table) node generate a line of python code. 

The order from SELECT ... statement is the order of items from the source table. And we won't generate the code according to this information.